cilium-sysdump/utils.py

```diff
@@ -165,15 +167,26 @@ def get_pods_status_iterator_by_labels(label_selector, host_ip_filter,
     # feature in the future. The following filter can be extended by
     # modifying the following kubectl custom-columns and the associated
     # grep command.
-    host_ip_filter_cmd = "kubectl get pods --no-headers " \
-        "-o=custom-columns=NAME:.metadata.name," \
-        "HOSTIP:.status.hostIP --all-namespaces | grep -E \"{}\" | " \
-        "awk '{{print $1}}'"
-    host_ip_filter_cmd = host_ip_filter_cmd.format(
-        "|".join(map(str, host_ip_filter)))
+    if node_filter and net.is_ipaddress(node_filter[0]):
+        log.info("filtering on node IP address ...")
+
+        node_filter_cmd = "kubectl get pods --no-headers " \
+            "-o=custom-columns=NAME:.metadata.name," \
+            "HOSTIP:.status.hostIP --all-namespaces | grep -E \"{}\" | " \
+            "awk '{{print $1}}'"
+    else:
```

Comment:
I think this else clause should only be reachable if `node_filter` is not empty, right?

Reply:
Good catch